packages/@aws-cdk/aws-iam/lib/policy-statement.ts

```diff
@@ -64,6 +64,10 @@ export class PolicyStatement {
   constructor(props: PolicyStatementProps = {}) {
     // Validate actions
     for (const action of [...props.actions || [], ...props.notActions || []]) {
+      if (cdk.Token.isUnresolved(action)) {
+        throw new Error('Cannot add \'Action\' that contains an unresolved token');
```

Comment:
I think what we want to do is only validate if we detect the value is NOT unresolved.

That way, we'll validate when we can and won't complain when we can't.

Add a test to check that we can pass in an unresolved value in a position we previously couldn't.

Reply:
Thank you. I think I have addressed your feedback. As an aside I'm newish to major contributions and wasn't sure how to `git` the changes up with out forcing (with lease) and that makes me feel bad so.... Any tips in that area would be greatly appreciated.

Reply:
:) yeah, I get that.

Here's the knowledge that will set you free: in our repository, you are allowed to not care about having a clean commit history on your PRs!

Because we will always squash-merge every PR into the main branch (collapsing it all into one commit, with the contents of the PR as the commit title & body), it doesn't actually matter whether your branch' commit history is clean or not. All that is going to go away anyway.

So I add commits named "oops" and "review" and "fix test" with wild abandon, and merge (without rebase!) from "master", and it's all good, **as long as** attention is paid to the PR body. 

Two additional notes:

* This happens to be true in our repository--might not be true in others, so I'm not sure how much this knowledge transfers to other projects (in case you're interested).
* In fact, for **GitHub** (not `git`) it's better to **never** force-push any branches, especially if they have review comments on them. It looks like GitHub remembers comments based on a combination of  something like *(commit hash, line number)*, and if the commit disappears because of force pushing, comments become orphaned and disappear from a thread.